crates/verify/src/bytecode.rs

```diff
@@ -0,0 +1,557 @@
+            println!("{}", serde_json::to_string_pretty(&output.clone().output())?);
+        }
+        Ok(output)
+    }
+}
+
+fn try_match(
+    local_bytecode: &[u8],
+    bytecode: &[u8],
+    constructor_args: &[u8],
+    match_type: &String,
+) -> Result<(bool, Option<String>)> {
+    // 1. Try full match
+    if match_type == "full" {
+        if local_bytecode.starts_with(bytecode) {
+            // Success => Full match
+            Ok((true, Some("full".to_string())))
+        } else {
+            // Failure => Try partial match
+            match try_partial_match(local_bytecode, bytecode, constructor_args) {
+                Ok(true) => Ok((true, Some("partial".to_string()))),
+                Ok(false) => Ok((false, None)),
+                Err(e) => Err(e),
+            }
+        }
+    } else {
+        match try_partial_match(local_bytecode, bytecode, constructor_args) {
+            Ok(true) => Ok((true, Some("partial".to_string()))),
+            Ok(false) => Ok((false, None)),
+            Err(e) => Err(e),
+        }
+    }
+}
+
+fn try_partial_match(
+    mut local_bytecode: &[u8],
+    mut bytecode: &[u8],
+    constructor_args: &[u8],
+) -> Result<bool> {
+    // 1. Check length of constructor args
+    if constructor_args.is_empty() {
+        // Assume metadata is at the end of the bytecode
+        local_bytecode = extract_metadata_hash(local_bytecode)?;
+        bytecode = extract_metadata_hash(bytecode)?;
+
+        // Now compare the creation code and bytecode
+        return Ok(local_bytecode.starts_with(bytecode));
+    }
+
+    // constructor_args in onchain bytecode
+    let (bytecode_args_range, onchain_args_at_end) =
+        constructor_args_range(bytecode, constructor_args);
+    // constructor_args in local bytecode
+    let (local_args_range, local_args_at_end) =
+        constructor_args_range(local_bytecode, constructor_args);
+    match (bytecode_args_range, local_args_range) {
+        (Some(onchain_range), Some(local_range)) => {
+            // Check if constructor args are at the end of the bytecode
+            if onchain_args_at_end {
+                bytecode = &bytecode[..onchain_range.start];
+            }
+            if local_args_at_end {
+                local_bytecode = &local_bytecode[..local_range.start];
+            }
+
+            // Note: If the constructor_args are not at the end of the bytecode we just assume that
+            // metadata hash is.
+
+            // Extract metdata from both
+            local_bytecode = extract_metadata_hash(local_bytecode)?;
+            bytecode = extract_metadata_hash(bytecode)?;
+
+            // Now compare the local code and bytecode
+            Ok(local_bytecode.starts_with(bytecode))
+        }
+        (None, None) => {
+            // Likely that this is runtime code. In that case extract_metadata and match
+            local_bytecode = extract_metadata_hash(local_bytecode)?;
+            bytecode = extract_metadata_hash(bytecode)?;
```

Comment:
We should pass a boolean arg marking if those are runtime codes instead of doing steps above to find out

Reply:
Added a `is_runtime` param to `try_match` and `try_partial_match`. Don't think we need a cli arg as we are checking both creation code and runtime code at the same.

Reply:
yeah, I meant function arg